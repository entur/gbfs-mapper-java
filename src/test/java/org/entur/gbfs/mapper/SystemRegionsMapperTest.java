package org.entur.gbfs.mapper;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.io.IOException;
import java.net.URL;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@ExtendWith({SnapshotExtension.class})
class SystemRegionsMapperTest {
    private static final ObjectMapper objectMapper = new ObjectMapper();
    private Expect expect;

    @SnapshotName("gbfs_v2_3_to_v3_0_system_regions_file_snapshot")
    @Test
    void testMapStationInformationFile() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v2_3/system_regions.json");
        org.entur.gbfs.v2_3.system_regions.GBFSSystemRegions testSubject = objectMapper.readValue(resource, org.entur.gbfs.v2_3.system_regions.GBFSSystemRegions.class);
        org.entur.gbfs.v3_0_RC2.system_regions.GBFSSystemRegions mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }

    @SnapshotName("gbfs_v3_0_to_v2_3_system_regions_file_snapshot")
    @Test
    void testMapStationInformationFileInverse() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v3_0-RC2/system_regions.json");
        org.entur.gbfs.v3_0_RC2.system_regions.GBFSSystemRegions testSubject = objectMapper.readValue(resource, org.entur.gbfs.v3_0_RC2.system_regions.GBFSSystemRegions.class);
        org.entur.gbfs.v2_3.system_regions.GBFSSystemRegions mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }
}