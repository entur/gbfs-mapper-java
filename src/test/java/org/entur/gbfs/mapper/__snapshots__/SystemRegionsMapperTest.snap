gbfs_v2_3_to_v3_0_system_regions_file_snapshot=[
  {
    "last_updated": "2021-12-30T17:59:23.000+00:00",
    "ttl": 86400,
    "version": "3.0",
    "data": {
      "regions": [
        {
          "region_id": "3",
          "name": [
            {
              "text": "North",
              "language": "en"
            }
          ]
        },
        {
          "region_id": "4",
          "name": [
            {
              "text": "East",
              "language": "en"
            }
          ]
        },
        {
          "region_id": "5",
          "name": [
            {
              "text": "South",
              "language": "en"
            }
          ]
        },
        {
          "region_id": "6",
          "name": [
            {
              "text": "West",
              "language": "en"
            }
          ]
        }
      ]
    }
  }
]


gbfs_v3_0_to_v2_3_system_regions_file_snapshot=[
  {
    "last_updated": 1640887163,
    "ttl": 86400,
    "version": "2.3",
    "data": {
      "regions": [
        {
          "region_id": "3",
          "name": "North"
        },
        {
          "region_id": "4",
          "name": "East"
        },
        {
          "region_id": "5",
          "name": "South"
        },
        {
          "region_id": "6",
          "name": "West"
        }
      ]
    }
  }
]